File: e-recode-room-BackEnd-RGD/src/main/java/com/erecode/erecoderoombackendrgd/service/UserManagementService.java
package com.erecode.erecoderoombackendrgd.service;

import com.erecode.erecoderoombackendrgd.model.Users;

import java.util.List;

public interface UserManagementService {

    List<Users> allUsers();

    Users register(Users users);

    Users login(String nic , String password);

}

File: e-recode-room-FrontEnd-RGD/src/app/service/UserManagementService.ts
import {HttpClient} from '@angular/common/http';
import {Injectable} from '@angular/core';
import {Users} from '../model/Users';
import {environment} from '../../environments/environment';

const URL = '/users';

@Injectable({
  providedIn: 'root'
})
export class UserManagementService {
  constructor(private http: HttpClient) {
  }

  login(nic: string, password: string) {
    return this.http.get<Users>(environment.backend_url + URL + '/login' + '/' + nic + '/' + password);
  }

  register(user: Users) {
    return this.http.post<Users>(environment.backend_url + URL + '/register', user);
  }
}
